internal/utils/utils.go

```diff
@@ -80,6 +78,26 @@ func GetTrustPolicy(ctx context.Context, role *iammanagerv1alpha1.Iamrole) (stri
 	return string(output), nil
 }
 
+func AppendOrReplaceTrustPolicyStatement(statements []iammanagerv1alpha1.TrustPolicyStatement, newStatements ...iammanagerv1alpha1.TrustPolicyStatement) []iammanagerv1alpha1.TrustPolicyStatement {
+	statementMap := make(map[string]int)
```

Comment:
shall we change this to `map[string]bool`? I think `bool` is easier to understand then index `i` or `len(statements)`

Reply:
> shall we change this to `map[string]bool`? I think `bool` as map value is easier to understand then index `i` and `len(statements)`

I use the index value, it's required to maintain the ordering otherwise we get random order and failed tests.